### DISCUSSION TITLE
Deploy docker problem

### DISCUSSION THREAD

**Discussion by dennisdewaard** [aahnik/tgcf#140] [Q&A] [answered]
Hello Guys,

I'm trying to deploy tgcf in docker but i can't get the enviroment variable config right.

I made a TGCF_CONFIG variable with the value: 

```
forwards: -source:-*********** dest:[-***********] -source:-*********** dest:[-***********] -source:-*********** dest:[-***********] show_forwarded_from:true
```

I get the error: 

```shell
tgcf.config.Config() argument after ** must be a mapping, not str
```

I think i'm doing something wrong but i can't get it right.

**Comment by tissole** [comment, marked as answer]
Check [this ](https://github.com/aahnik/tgcf/wiki/How-to-configure-tgcf-%3F)wiki for the correct syntax 
I think should be like this

```
forwards:
  - source: -source1
    dest: [-dest1,-dest2]
```

Try like this and see if is working.

**Comment by aahnik** [comment]
hi, @dennisdewaard, I hope you have read the [Install and run using docker ](https://github.com/aahnik/tgcf/wiki/Install-and-run-using-docker) page in the wiki.

The [`tgcf.config.yml`](https://github.com/aahnik/tgcf/wiki/How-to-configure-tgcf-%3F#example-configuration) is a multi-line file. You must follow the correct syntax of the [yaml](https://github.com/aahnik/tgcf/wiki/How-to-configure-tgcf-%3F#introducing-yaml) file. the answer given by @tissole  is correct.

Some extra things to note while using `docker run`

Passing the configuration inside the `TGCF_CONFIG` env var is problematic. if you are using, the `--env-file=.env` option while `docker run`.

**You should rather create a file called `tgcf.config.yml` and then mount that into the docker container.**

Example, 

if the absolute path to `tgcf.config.yml` in your computer is: `/home/user/folder/tgcf.config.yml`

Then you should use this docker run command:

```shell
docker run -v home/user/folder/tgcf.config.yml:/app/tgcf.config.yml --env-file=.env -it aahnik/tgcf
```

**Comment by dennisdewaard** [comment]
Thanks Guys!

I just this one because synology docker has some trouble with a multi-line environment.

**You should rather create a file called tgcf.config.yml and then mount that into the docker container.**

It's working great now!

**Comment by dennisdewaard** [comment]
Got an other error and it looks like it is going wrong with eddited messages:

```shell
ERROR:telethon.client.updates:Unhandled exception on edited_message_handler
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/telethon/client/updates.py", line 454, in _dispatch_update
    await callback(event)
  File "/usr/local/lib/python3.9/site-packages/tgcf/live.py", line 88, in edited_message_handler
    message = apply_plugins(message)
  File "/usr/local/lib/python3.9/site-packages/tgcf/plugins.py", line 38, in apply_plugins
    message = plugin.modify(message)
  File "/usr/local/lib/python3.9/site-packages/tgcf_filter/__init__.py", line 48, in modify
    if self.text_safe(message):
  File "/usr/local/lib/python3.9/site-packages/tgcf_filter/__init__.py", line 56, in text_safe
    for forbidden in flist.blacklist:
TypeError: 'NoneType' object is not iterable
```

**Reply by dennisdewaard** [reply]
Sorry It's working now.
Always use the latest version with docker xD.....
